Parse/Internal/Object/BatchController/PFObjectBatchController.m

```diff
@@ -99,9 +103,84 @@ - (BFTask *)_processFetchResultAsync:(NSDictionary *)result forObjects:(NSArray
+        NSMutableArray *tasks = [NSMutableArray arrayWithCapacity:objectBatches.count];
+        for (NSArray *batch in objectBatches) {
+            PFRESTCommand *command = [self _deleteCommandForObjects:batch withSessionToken:sessionToken];
+            BFTask *task = [[self.dataSource.commandRunner runCommandAsync:command
+                                                              withOptions:PFCommandRunningOptionRetryIfFailed] continueWithSuccessBlock:^id(BFTask *task) {
+                PFCommandResult *result = task.result;
+                return [self _processDeleteResultsAsync:[result result] forObjects:batch];
+            }];
+            [tasks addObject:task];
+        }
+        return [[BFTask taskForCompletionOfAllTasks:tasks] continueWithBlock:^id(BFTask *task) {
+            NSError *taskError = task.error;
+            if (taskError && [taskError.domain isEqualToString:BFTaskErrorDomain]) {
+                NSArray *taskErrors = taskError.userInfo[@"errors"];
+                NSMutableArray *errors = [NSMutableArray array];
+                for (NSError *error in taskErrors) {
+                    if ([error.domain isEqualToString:BFTaskErrorDomain]) {
+                        [errors addObjectsFromArray:error.userInfo[@"errors"]];
+                    } else {
+                        [errors addObject:error];
+                    }
+                }
+                return [BFTask taskWithError:[NSError errorWithDomain:BFTaskErrorDomain
+                                                                 code:kBFMultipleErrorsError
+                                                             userInfo:@{ @"errors" : errors }]];
+            }
+            return task;
+        }];
+    }] continueWithSuccessResult:objects];
+}
+
+- (PFRESTCommand *)_deleteCommandForObjects:(NSArray *)objects withSessionToken:(NSString *)sessionToken {
+    NSMutableArray *commands = [NSMutableArray arrayWithCapacity:objects.count];
+    for (PFObject *object in objects) {
+        PFRESTCommand *deleteCommand = [PFRESTObjectCommand deleteObjectCommandForObjectState:object._state
+                                                                             withSessionToken:sessionToken];
+        [commands addObject:deleteCommand];
+    }
+    return [PFRESTObjectBatchCommand batchCommandWithCommands:commands sessionToken:sessionToken];
+}
+
+- (BFTask *)_processDeleteResultsAsync:(NSArray *)results forObjects:(NSArray *)objects {
+    NSMutableArray *tasks = [NSMutableArray arrayWithCapacity:results.count];
+    [results enumerateObjectsUsingBlock:^(NSDictionary *result, NSUInteger idx, BOOL *stop) {
+        PFObject *object = objects[idx];
+        NSDictionary *errorResult = result[@"error"];
+        NSDictionary *successResult = result[@"success"];
+
+        id<PFObjectControlling> controller = [[object class] objectController];
```

Comment:
Don't we have access to the data source here? Why go through the class dispatch method.